core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -96,7 +108,20 @@ class TransactionStateManager(brokerId: Int,
   private[transaction] val transactionMetadataCache: mutable.Map[Int, TxnMetadataCacheEntry] = mutable.Map()
 
   /** number of partitions for the transaction log topic */
-  private val transactionTopicPartitionCount = getTransactionTopicPartitionCount
+  private var _transactionTopicPartitionCount: Option[Int] = Option.empty // lazy, once-only evaluation
```

Comment:
I think this should probably be `@volatile` since the reads are not synchronized. Maybe we can use `AtomicReference` or some future thing here?